doc/doxygen/src/gnrc_ipv6_auto_subnets-flow.puml

```diff
@@ -0,0 +1,31 @@
+' to generate SVG run plantuml -tsvg gnrc_ipv6_auto_subnets-flow.puml
+@startuml
+
+participant "2e:a3:9e:a9:68:**23**" as A
+participant "2e:a3:9e:a9:68:**f6**" as B
+participant "2e:a3:9e:a9:68:**42**" as C
```

Comment:
Does the least significant byte of the address being bold have any meaning? If so, this meaning should be apparent within the graphic somehow (a graphic should be able to stand on its own). Otherwise, it should be removed, as I think it confuses more than it helps understanding the algorithm.

Reply:
I thought it's a nice visual aid to see what is the difference between the l2 addresses.

Reply:
The problem to me was more that they are highlighted in general. Now that I _know_ what this is about, my perspective shifted, so I'm not sure if making them italic makes any difference. I think that L2 addresses are different should be assumed, so highlighting the difference seems somewhat redundant to me, however, still. Do they have to be literal addresses? Can't they just be A, B, C?

Reply:
The idea was to visualize that the order in which the prefixes are chosen / where each router starts counting it's prefixes depends on the order of the l2 addresses.
I hoped changing only the last byte and highlighting that would make it easier to understand the concept.

Reply:
> The idea was to visualize that the order in which the prefixes are chosen / where each router starts counting it's prefixes depends on the order of the l2 addresses.
> I hoped changing only the last byte and highlighting that would make it easier to understand the concept.

Mh, I think by abstracting this more to numeric or alphanumeric abstract L2 addresses would help to highlight this concept more. How about something like "L2 addr. 1", "L2 addr. 2", ...

Reply:
Because, just from looking at the graphic alone, this was not clear to me... but mmh, not sure "L2 addr. 1", ... would elevate that either.... Let me ponder on that.

Reply:
Can we go "Node A", "Node B", ... again and add a `<` in between them to make the ordering between them more clearer?

Reply:
[Sequence Diagram](https://plantuml.com/en/sequence-diagram) doesn't allow for text between participants though

Reply:
Maybe some free floating text? :-/

Reply:
How about this

![gnrc_ipv6_auto_subnets-flow](https://user-images.githubusercontent.com/1301112/133761685-fef02b8d-691b-4b5f-93ab-41c6affb9ee5.png)

Reply:
If the order of their significance now would also be reflected in the order they are shown (so C in the second column, B in the third), I'd be happy :-).

Reply:
(of course feel free to adjust the labels then as well).

Reply:
> If the order of their significance now would also be reflected in the order they are shown

I thought mixing them up made it clearer that it's not the order of appearance that matters but the order of addresses - but alright, I can shuffle them around.

Reply:
![](https://raw.githubusercontent.com/RIOT-OS/RIOT/750fbe5f3e76a7bf8f11a54d7c9b31ea7df92d0c/doc/doxygen/src/gnrc_ipv6_auto_subnets-flow.svg)

Reply:
> > If the order of their significance now would also be reflected in the order they are shown
> 
> I thought mixing them up made it clearer that it's not the order of appearance that matters but the order of addresses - but alright, I can shuffle them around.

No, mixing things up to make them clearer, rarely works ;-)